pkg/datasource/sql/undo/base/undo.go

```diff
@@ -263,10 +275,12 @@ func (m *BaseUndoLogManager) Undo(ctx context.Context, dbType types.DBType, xid
 		undoLogRecords = append(undoLogRecords, record)
 	}
 
+	var exists bool
 	for _, record := range undoLogRecords {
+		exists = true
 		if !record.CanUndo() {
 			log.Infof("xid %v branch %v, ignore %v undo_log", record.XID, record.BranchID, record.LogStatus)
-			continue
+			return nil
```

Comment:
If one of the multiple undologs does not mark undo, the match will be terminated. However, in the process executed by the branch transaction, undo should be isolated from each other